mvpa2/misc/plot/scatter.py

```diff
@@ -392,13 +395,21 @@ def jitter_me(x, w):
     #       actually consider masked-out values
     if ax_hist_x is not None:
         ax_hist_x.xaxis.set_major_formatter(nullfmt)
-        histx = ax_hist_x.hist(x, bins=binsx, facecolor='b')
+        if mask is not None:
+            histx = ax_hist_x.hist(x_masked, bins=binsx, facecolor='b')
+        else:
+            histx = ax_hist_x.hist(x, bins=binsx, facecolor='b')
```

Comment:
so it could be a 1 liner
```python
histx = ax_hist_x.hist(x if mask is None else x_masked , bins=binsx, facecolor='b')
```
here and below.  duplication is evil!